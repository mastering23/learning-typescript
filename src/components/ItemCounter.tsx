interface Props {
  name: string;
  quantity?: number;
}
export const ItemCounter = ({ name, quantity }: Props) => {

  const handleClick =() =>{
      console.log(`click  en ${name}`);
  }


  return (
    <section
      style={{
        display: 'flex',
        alignItems: 'center',
        gap: 10,
      }}>

      <span style={{
        width: 150,
        marginTop: 20,
      }}>{name}</span>
     
      <button
        onClick={handleClick}>+1</button>
      <span>{quantity}</span>
      
      <button>-1</button>
    </section>

  );

}