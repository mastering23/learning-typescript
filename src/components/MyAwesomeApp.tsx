import type { CSSProperties } from "react";

export const MyAwesomeApp = () => {

  const name1 = "David";
  const name2 = "Bruno";

  const foodTopics = ['Chesses', 'Tomato', 'Salsa', 'Pinapple'];

  const isActive = true;

  const address = {
    zipCode: '10040',
    country: 'USA',
  }

  const myStyles: CSSProperties = {
    backgroundColor: '#9c1515ff',
    color: 'white',
    borderRadius: 10,
    padding: 20,

  }

  return (
    <div>
      <h1>Fernando</h1>
      <h3>Herrera</h3>

      {/* Variables  */}
      <h2>{name1}</h2>
      <h2>{name2}</h2>

      {/* suma */}
      {2 + 2}

      {/* arreglos  */}
      <p>{foodTopics.join(', ')}</p>

      {/* ternarios  */}
      <h2>{isActive ? 'Activo' : 'No Activo'}</h2>

      {/* object  + css */}
      <p
        style={myStyles}
      >

        {JSON.stringify(address)}

      </p>

    </div>

  );

}