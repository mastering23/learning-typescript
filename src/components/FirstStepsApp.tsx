import { MyAwesomeApp } from "./MyAwesomeApp";
import { ItemCounter} from "./itemCounter";

interface ItemInCart{
  productName: string;
  quantity: number;
}

const intemsIncart : ItemInCart [] = [
{productName: 'PS 5',quantity:20},
{productName: 'Xbox',quantity:10},
{productName: 'Nintendo',quantity:16},

]

export function FirstStepsApp(){

  return(
    <>
     <h1>Hello wolrd</h1>
     <p>esto es parrafo</p>
     <button>Click me </button>
     <MyAwesomeApp />

     {intemsIncart.map(({productName,quantity})=>(
      <ItemCounter key={productName} name ={productName}quantity={quantity} />
     ))}
    {/* <ItemCounter name ="PS 5" quantity={20}/>
    <ItemCounter name = "Xbox" quantity={10}/>
    <ItemCounter name = "Nitendo"quantity={16}/> */}
    </>

  );

}
