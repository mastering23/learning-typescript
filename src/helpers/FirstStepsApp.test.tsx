import { describe, expect, test } from "vitest";
import { FirstStepsApp } from "../components/FirstStepsApp";
import { render } from "@testing-library/react";

describe('FirstStepsApp',()=>{
test('Should match snapshot',()=>{
  const {container} = render (<FirstStepsApp />);

  expect(container).toMatchSnapshot();

  
})
  
});