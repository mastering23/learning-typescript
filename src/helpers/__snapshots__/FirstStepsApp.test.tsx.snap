// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepsApp > Should match snapshot 1`] = `
<div>
  <h1>
    Hello wolrd
  </h1>
  <p>
    esto es parrafo
  </p>
  <button>
     Click me 
  </button>
  <div>
    <h1>
      Fernando
    </h1>
    <h3>
      Herrera
    </h3>
    <h2>
      David
    </h2>
    <h2>
      Bruno
    </h2>
    4
    <p>
      Chesses, Tomato, Salsa, Pinapple
    </p>
    <h2>
      Activo
    </h2>
    <p
      style="background-color: rgb(156, 21, 21); color: white; border-radius: 10px; padding: 20px;"
    >
      {"zipCode":"10040","country":"USA"}
    </p>
  </div>
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
  <div
    data-testid="ItemCounter"
  />
</div>
`;
